sdks/python/apache_beam/runners/interactive/display/display_manager.py

```diff
@@ -50,54 +50,49 @@ def _formatter(string, pp, cycle):  # pylint: disable=unused-argument
                pipeline_graph_renderer):
     """Constructor of DisplayManager.
 
     Args:
-      pipeline_info: (interactive_runner.PipelineInfo)
       pipeline_proto: (Pipeline proto)
-      caches_used: (set of str) A set of PCollection IDs of those whose cached
-          results are used in the execution.
+      pipeline_analyzer: (PipelineAnalyzer)
       cache_manager: (interactive_runner.CacheManager) DisplayManager fetches
           the latest status of pipeline execution by querying cache_manager.
-      referenced_pcollections: (dict from str to PCollection proto) PCollection
-          ID mapped to PCollection referenced during pipeline execution.
-      required_transforms: (dict from str to PTransform proto) mapping from
-          transform ID to transforms that leads to visible results.
       pipeline_graph_renderer: (pipeline_graph_renderer.PipelineGraphRenderer)
           decides how a pipeline graph is rendered.
     """
     # Every parameter except cache_manager is expected to remain constant.
+    self._analyzer = pipeline_analyzer
     self._cache_manager = cache_manager
-    self._referenced_pcollections = referenced_pcollections
     self._pipeline_graph = interactive_pipeline_graph.InteractivePipelineGraph(
         pipeline_proto,
-        required_transforms=required_transforms,
-        referenced_pcollections=referenced_pcollections,
-        cached_pcollections=caches_used)
+        required_transforms=self._analyzer.tl_required_trans_ids(),
+        referenced_pcollections=self._analyzer.tl_referenced_pcoll_ids(),
+        cached_pcollections=self._analyzer.caches_used())
     self._renderer = pipeline_graph_renderer
 
     # _text_to_print keeps track of information to be displayed.
     self._text_to_print = collections.OrderedDict()
     self._text_to_print['summary'] = (
         'Using %s cached PCollections\nExecuting %s of %s '
         'transforms.') % (
-            # TODO(qinyeli): required_transforms includes ReadCache and
-            # WriteCache fix it.
-            len(caches_used), len(required_transforms),
+            len(self._analyzer.caches_used()),
+            (len(self._analyzer.tl_required_trans_ids())
+             - len(self._analyzer.read_cache_ids())
+             - len(self._analyzer.write_cache_ids())),
             len(pipeline_proto.components.transforms[
                 pipeline_proto.root_transform_ids[0]].subtransforms))
-    self._text_to_print.update(
-        {pcoll_id: "" for pcoll_id in referenced_pcollections})
+    self._text_to_print.update({
+        pcoll_id: "" for pcoll_id
+        in self._analyzer.tl_referenced_pcoll_ids()})
 
     # _pcollection_stats maps pcoll_id to
     # { 'cache_label': cache_label, version': version, 'sample': pcoll_in_list }
     self._pcollection_stats = {}
-    for pcoll_id in pipeline_info.all_pcollections():
+    for pcoll_id in self._analyzer.tl_referenced_pcoll_ids():
```

Comment:
Why do we go from checking all PCs to checking only top-level-referenced ones?

Reply:
The PCollections are filtered around line 140 by referenced_pcollection, which contains only top_level ones.
The non-top level ones were added to _pcollection_stats but not used anywhere.